cgroups/fs/apply_raw.go

```diff
@@ -88,6 +91,93 @@ func Apply(c *cgroups.Cgroup, pid int) (map[string]string, error) {
 	return paths, nil
 }
 
+// Stop stops all processes in the current cgroup
+func Stop(c *cgroups.Cgroup) error {
+	d, err := getCgroupData(c, 0)
+	if err != nil {
+		return err
+	}
+
+	// first we send SIGTERM to all tasks inside the cgroup
+	if err := Kill(c, syscall.SIGTERM); err != nil {
+		return err
+	}
+
+	// if failed to stop with only SIGTERM, we send SIGKILL
+	if err := waitStop(d, 100*time.Millisecond, 5*time.Second); err != nil {
+		if err := Kill(c, syscall.SIGKILL); err != nil {
+			return err
+		}
+		return waitStop(d, 100*time.Millisecond, -1*time.Second)
+	}
+	return nil
+}
+
+// Kill sends the signal to all processes in the current cgroup.
+// This assumes that the freezer cgroup is setup. It first freeze the cgroup,
+// send the signal to all processes inside the cgroup and then unfreeze it to
+// make sure that signal is sent to the processes before it has a chance to
+// do something else
+func Kill(c *cgroups.Cgroup, signal syscall.Signal) error {
+	Freeze(c, cgroups.Frozen)
```

Comment:
Hmm, right. The original assumption in the spec is that freezer should be available ( also actually devices since `GetPids` code use it ). I think if `freezer` is not available, then we would have to:
- pick any one of the available cgroup in `subsystems` and recursively send signal to all of its processes
- and/or create our own named cgroup just for process tracking